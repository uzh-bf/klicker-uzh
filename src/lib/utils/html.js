import sanitizeHTML from 'sanitize-html'
import { stateToHTML } from 'draft-js-export-html'

function toSanitizedHTML(contentState) {
  try {
    // extract the current draftjs content state
    return (
      contentState
      |> stateToHTML // convert the content state to HTML
      |> sanitizeHTML // sanitize the resulting HTML
    )
  } catch (e) {
    console.error(`Failed sanitizing HTML... ${e}`)
  }

  return null
}

export { toSanitizedHTML }
