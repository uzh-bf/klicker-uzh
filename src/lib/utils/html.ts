import sanitizeHTML from 'sanitize-html'
import { stateToHTML } from 'draft-js-export-html'

export function toSanitizedHTML(contentState): string {
  try {
    // extract the current draftjs content state
    return sanitizeHTML(stateToHTML(contentState))
  } catch (e) {
    console.error(`Failed sanitizing HTML... ${e}`)
  }

  return null
}
