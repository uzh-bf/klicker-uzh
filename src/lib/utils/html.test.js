import { ContentState } from 'draft-js'
import { toSanitizedHTML } from './html'

describe('toSanitizedHTML', () => {
  it('works', () => {
    // initialize a content state
    const contentState = ContentState.createFromText(
      '<script src="bla" /><br /><strong>Hello World</strong><br /><img src="blu" /><link rel="bla" />'
    )

    // use sanitizing function
    const sanitized = toSanitizedHTML(contentState)

    expect(sanitized).toMatchSnapshot()
  })
})
