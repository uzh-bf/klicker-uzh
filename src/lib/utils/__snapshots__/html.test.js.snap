// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toSanitizedHTML works 1`] = `"<p>&lt;script src=&quot;bla&quot; /&gt;&lt;br /&gt;&lt;strong&gt;Hello World&lt;/strong&gt;&lt;br /&gt;&lt;img src=&quot;blu&quot; /&gt;&lt;link rel=&quot;bla&quot; /&gt;</p>"`;
